src/main.rs

```diff
@@ -44,48 +44,56 @@ fn main() {
     exit(result);
 }
 
-/// Try to read `shell.nix` from the current working dir.
-fn get_shell_nix(shellfile: &PathBuf) -> Result<NixFile, ExitError> {
+/// Reads a nix filename given by the user and either returns
+/// the `NixFile` type or exists with a helpful error message
+/// that instructs the user how to write a minimal `shell.nix`.
+fn find_nix_file(shellfile: &PathBuf) -> Result<NixFile, ExitError> {
     // use shell.nix from cwd
-    Ok(NixFile::from(locate_file::in_cwd(&shellfile).map_err(
-        |_| {
+    Ok(NixFile::from(
+        // `::in_cwd` is guaranteed to return an absolute path
```

Comment:
Shouldn't `in_cwd` just return an `AbsPathBuf` in that case?

Reply:
Good idea, changed in last commit.